include/utils/tree.h

```diff
@@ -76,16 +76,19 @@ class Tree : public TreeBase
 
   /**
    * @returns a pointer to the element containing point p,
+   * for elements of dimension elem_dim,
    * optionally restricted to a set of allowed subdomains,
    * optionally using a non-zero relative tolerance for searches.
    */
-  const Elem* find_element(const Point& p,
-                           const std::set<subdomain_id_type>
-                           *allowed_subdomains = NULL,
-                           Real relative_tol = TOLERANCE) const;
```

Comment:
We're removing the original signature of the function?

This is a public function; although I doubt anyone's directly using it I'd rather not break the API just in case.  Could you add the elem_dim version as a separate overload again?